ptrauth module include issue for Swift 6.1 release

On Swift 6.1 release, the LLVM dependency has included the following commit.

https://github.com/swiftlang/llvm-project/commit/0481f049c37029d829dbc0c0cc5d1ee71c6d1c9a

And it added `ptrauth.h` header to clang's modulemap file.

```
module ptrauth {
  header "ptrauth.h"
  export *
}
```

<img width="1873" alt="Image" src="https://github.com/user-attachments/assets/ed3a273d-7141-4d42-a0c0-467b08491e77" />

And on swift-corelibs-foundation repo, we have the following statements between a `CF_EXTERN_C_BEGIN` and `CF_EXTERN_C_END` block.

```
#if __has_include(<ptrauth.h>)
#include <ptrauth.h>
#endif
```

It will end up causing include issue for the CoreFoundation on Linux platform.

```
/usr/lib/swift/CoreFoundation/CFBase.h:683:1: error: import of C++ module 'ptrauth' appears within extern "C" language linkage specification [-Wmodule-import-in-extern-c]
```

The fix is move such include statements out of `CF_EXTERN_C_BEGIN` and `CF_EXTERN_C_END` block.

--- a/Sources/CoreFoundation/include/CFBase.h
+++ b/Sources/CoreFoundation/include/CFBase.h
@@ -669,6 +669,8 @@ CF_IMPLICIT_BRIDGING_DISABLED
 CF_EXPORT
 CFTypeRef CFMakeCollectable(CFTypeRef cf) CF_AUTOMATED_REFCOUNT_UNAVAILABLE;
 
+CF_EXTERN_C_END
+
 #if DEPLOYMENT_RUNTIME_SWIFT
 
 #if TARGET_RT_64_BIT
@@ -695,8 +697,6 @@ CFTypeRef CFMakeCollectable(CFTypeRef cf) CF_AUTOMATED_REFCOUNT_UNAVAILABLE;
 #else
 #define __ptrauth_cf_objc_isa_pointer
 #endif
-
-CF_EXTERN_C_END
 
 #endif /* ! __COREFOUNDATION_CFBASE__ */
 

